src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1027 @@ const ActorInit En_Js_InitVars = {
+    this->maskDlSelector = 0;
+    this->actor.terminalVelocity = -9.0f;
+    this->actor.gravity = -1.0f;
+
+    cs = this->actor.cutscene;
+    i = 0;
+    while (i != 2) {
+        this->cutscenes[i] = cs;
+        if (cs != -1) {
+            this->actor.cutscene = cs;
+            cs = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        }
+        i++;
+    }
+
+    this->cutsceneIndex = -1;
+
+    switch (ENJS_GET_TYPE(&this->actor)) {
+        case 0:
+            this->maskDlSelector = 0;
+            this->actionFunc = func_8096A6F4;
+
+            Animation_PlayLoop(&this->skelAnime, &gMoonChildSittingAnim);
+            func_8016566C(0x3C);
+
+            if (gSaveContext.save.weekEventReg[84] & 0x20) {
+                Inventory_DeleteItem(0x35, gItemSlots[0x35]);
+                gSaveContext.save.weekEventReg[84] &= (u8)~0x20;
+                break;
+            }
+        default:
+            break;
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+            this->maskDlSelector = ENJS_GET_TYPE(&this->actor);
+            func_80968B8C(this, play);
+            this->actionFunc = func_80969B5C;
+            func_80968A5C(this);
+            if (func_809692A8(ENJS_GET_TYPE(&this->actor) + 4)) {
+                Actor_MarkForDeath(&this->actor);
+                break;
+            }
+            break;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            this->maskDlSelector = ENJS_GET_TYPE(&this->actor) - 4;
+            this->actionFunc = func_8096A104;
+            break;
+    }
+}
+
+void EnJs_Destroy(Actor* thisx, PlayState* play) {
+    u32 paramsF;
+    EnJs* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+
+    paramsF = ENJS_GET_TYPE(&this->actor);
+    switch (paramsF) {
+        case 0:
+            func_80165690();
+            return;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            if (!func_809692A8(paramsF)) {
+                func_80969400(ENJS_GET_TYPE(&this->actor));
+            }
+        default:
+            return;
+    }
+}
+
+void func_80968A5C(EnJs* this) {
+    if (Rand_ZeroOne() < 0.5f) {
+        Animation_MorphToLoop(&this->skelAnime, &gMoonChildRunningAnim, -10.0f);
+        this->unk_2B4 = 5.0f;
+    } else {
+        Animation_MorphToLoop(&this->skelAnime, &gMoonChildWalkingAnim, -10.0f);
+        this->unk_2B4 = 2.5f;
+    }
+    this->unk_2B8 &= ~4;
+    this->unk_2BC = Rand_ZeroFloat(40.0f) + 80.0f;
+}
+
+void func_80968B18(EnJs* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gMoonChildStandingAnim, -10.0f);
+    this->unk_2B4 = 0.0f;
+    this->unk_2BC = Rand_ZeroFloat(20.0f) + 40.0f;
+    this->unk_2B8 |= 4;
+}
+
+s32 func_80968B8C(EnJs* this, PlayState* play) {
+    Path* path;
+    f32 f0;
+    f32 f2;
+    s32 params;
+    s32 i;
+    f32 sp18;
+    Vec3s* phi_a0;
+
+    sp18 = 0.0f;
+
+    params = ENJS_GET_PATH_INDEX(&this->actor) >> 0xA;
```

Comment:
```suggestion
    params = ENJS_GET_PATH_INDEX(&this->actor);
```
The shift should still be in the macro